internal/translator/translate.go

```diff
@@ -48,23 +49,28 @@ func IngressRouteToHTTPProxy(ir *irv1beta1.IngressRoute) (*hpv1.HTTPProxy, []str
 
 	// TODO(youngnick): Investigate if we should skip logically empty IngressRoutes
 
+	if ir.Spec.VirtualHost == nil {
+		return nil, warnings, errors.New("unimplemented: Can't translate non-root IngressRoutes yet")
+	}
 	hp.Spec.VirtualHost = ir.Spec.VirtualHost
```

Comment:
I suggest collecting all the parts of the hp that you need first, then construct it. That way you know that once `hp` is initialised, it is complete and no further work is required.  Structurally move line 32 down to ~ 57 and you can construct and return the hp in one hit.